packages/app/src/systems/Account/machines/accountsMachine.tsx

```diff
@@ -260,6 +293,39 @@ export const accountsMachine = createMachine(
           await CoreService.clear();
         },
       }),
+      addAccount: FetchMachine.create<never, Account>({
+        showError: true,
+        maxAttempts: 1,
+        async fetch() {
+          const name = await AccountService.generateAccountName();
+
+          if (await AccountService.checkAccountNameExists(name)) {
+            throw new Error('Account name already exists');
+          }
+
+          // Add account to vault
+          const accountVault = await VaultService.addAccount({
+            // TODO: remove this when we have multiple vaults
+            // https://github.com/FuelLabs/fuels-wallet/issues/562
+            vaultId: 0,
+          });
+
+          // Add account to the database
+          let account = await AccountService.addAccount({
+            data: {
+              name,
+              ...accountVault,
+            },
+          });
+
+          // set as active account
+          account = await AccountService.setCurrentAccount({
+            address: account.address.toString(),
+          });
+
+          return account;
+        },
+      }),
```

Comment:
We had another machine for this before, to avoid the action to be blocked if the account list is loading.

Reply:
hmm, okay, I didn't think of that, I thought we had a AddAccount machine because we had a AddAccount page for it.

Reply:
Going to move this back now.